modules/core/src/main/scala/org/tessellation/l0/SnapshotService.scala

```diff
@@ -0,0 +1,59 @@
+package org.tessellation.l0
+
+import cats.Applicative
+import cats.effect.kernel.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.config.types.SnapshotConfig
+import org.tessellation.dag.block.{BlockStorage, BlockValidator}
+import org.tessellation.schema.height.Height
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosLong
+import io.estatico.newtype.ops._
+
+trait SnapshotService[F[_]] {
+  def createSnapshot(minTipHeight: PosLong): F[DAGSnapshot]
+}
+
+object SnapshotService {
+
+  def make[F[_]: Applicative: Async](
+    config: SnapshotConfig,
+    blockStorage: BlockStorage[F],
+    blockValidator: BlockValidator[F],
+    snapshotStorage: SnapshotStorage[F]
+  ): SnapshotService[F] =
+    new SnapshotService[F] {
+
+      def createSnapshot(minTipHeight: PosLong): F[DAGSnapshot] =
+        for {
+          previousSnapshotHeight <- snapshotStorage.getLastSnapshotHeight
+          nextSnapshotHeight = Height(previousSnapshotHeight.coerce + config.heightInterval)
+
+          // In 1.0: validateSnapshotHeightIntervalCondition
+          //  if (minTipHeight > nextHeight + config.heightDelayInterval)
```

Comment:
I am wondering if we should check it here

Reply:
Especially that we are not sure how the trigger will work. Old implementation was based on tips used up by the node itself. Not sure how the trigger will work in L0 (which not necessarily will create it's own blocks).

Reply:
We should be able to reuse essentially the same logic from the Constellation repo i.e. using tip height. However in order to promote separation of the L0 from L1 or rather the snapshot trigger from block creation, what if we updated tip height as blocks are received rather than from the checkpoint storage cache?